app/Jobs/ImportPipeline/ImportInstagram.php

```diff
@@ -101,7 +101,11 @@ public function handle()
 
                 $path = storage_path("app/$importData->path");
                 $storagePath = "public/m/{$monthHash}/{$userHash}";
-                $newPath = "app/$storagePath/$filename";
+                $dir = "app/$storagePath";
+                if(!is_dir(storage_path($dir))) {
+                    mkdir(storage_path($dir), 0777, true);
```

Comment:
isn't 0777 a bit overkill though ?
```suggestion
                    mkdir(storage_path($dir), umask(), true);
```

just a thought,
new comer here looking around :)
I'd make sure it's not more than 0750 if hosting myself.

Reply:
The way I read the [documentation of mkdir](https://www.php.net/manual/en/function.mkdir.php), the mode parameter is already modified by the current umask. Also, 0777 is the default argument if no other is supplied. I quickly tested it from the command line, and with umask set to 022, `mkdir("dir", 0777)` creates a directory with mode 0755.

But sure, 0750 would work, too. Maybe the maintainers could state their preference.